adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+import random
+
+NUM_DRAWN = 10
+
 def draw_letters():
-    pass
+    '''
+    Creates a list of specific length. Elements of the list are randomly
+    selected letters from a pool defined within the function.
+
+    Returns:
+        letters_drawn (list): random letters in a list of X length
+
+    '''
+    letters_drawn = []
+    letter_pool = build_letter_pool()
+
+    for _ in range(NUM_DRAWN):
+        random_index = random.randint(0, len(letter_pool) - 1)
+        selected_letter = letter_pool[random_index]
+
+        letters_drawn.append(selected_letter)
+        letter_pool.remove(selected_letter)
+    
+    return letters_drawn
 
 def uses_available_letters(word, letter_bank):
-    pass
+    '''
+    Determines if a given word can be spelled using only letters from
+    the letter bank.
+
+    Checks whether each letter in the word is found in the letter bank,
+    removing letters as they are found.
+
+    Parameters:
+        word (str): The word to be checked.
+        letter_bank (list): List of letters available to form words.
+
+    Returns:
+        bool: True if the given word can be made using only the letters
+        available in letter_bank, otherwise False
+    '''
+    copied_list = []
+
+    for letter in letter_bank:
+        copied_list.append(letter)
+
+    for letter in word.upper():
+        if letter in copied_list:
+            copied_list.remove(letter)
+        else:
+            return False
+
+    return True
 
 def score_word(word):
-    pass
+    '''
+    Calculates the total score of the given word based on letter values.
 
+    Each letter has a specific assigned point value. The total score is
+    the sum of these values. Words of a minimum length receive an 
+    additional bonus.
+    
+    Parameters:
+        word (str): The word to be scored.
+
+    Returns:
+        total_score (int): The total score for the word, including any
+        applicable bonuses.
+    '''
+    POINT_VALUES  = {
+    'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4, 
+    'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3, 
+    'Q': 10, 'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8, 
+    'Y': 4, 'Z': 10
+}
+    BONUS_LENGTH = 7
+    BONUS_POINTS = 8
+
+    total_score = sum(POINT_VALUES.get(letter.upper(), 0) for letter in word)
+    
+    if len(word) >= BONUS_LENGTH:
+        total_score += BONUS_POINTS
```

Comment:
While a hand can't be more than 10 letters and therefore a word can't be more than 10 letters, you could make this check more explicit and conform to the instructions in the README more by having the check on line 81 be: `if BONUS_LENGTH <= len(word) <= 10:` 

And put that 10 in a constant too.